test_conformance/api/test_api_consistency.cpp

```diff
@@ -0,0 +1,479 @@
+                      "CL_DEVICE_QUEUE_ON_DEVICE_PROPERTIES returned a "
+                      "non-zero value\n");
+            return TEST_FAIL;
+        }
+
+        // clGetDeviceInfo, passing
+        // CL_DEVICE_QUEUE_ON_DEVICE_PREFERRED_SIZE,
+        // CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE,
+        // CL_DEVICE_MAX_ON_DEVICE_QUEUES, or
+        // CL_DEVICE_MAX_ON_DEVICE_EVENTS
+        // Returns 0 if device does not support Device-Side Enqueue and
+        // On-Device Queues.
+
+        cl_uint u = 0;
+
+        error =
+            clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_DEVICE_PREFERRED_SIZE,
+                            sizeof(u), &u, NULL);
+        test_error(error,
+                   "Unable to query CL_DEVICE_QUEUE_ON_DEVICE_PREFERRED_SIZE");
+        if (u != 0)
+        {
+            log_error("CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES returned 0 but "
+                      "CL_DEVICE_QUEUE_ON_DEVICE_PREFERRED_SIZE returned a "
+                      "non-zero value\n");
+            return TEST_FAIL;
+        }
+
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE,
+                                sizeof(u), &u, NULL);
+        test_error(error, "Unable to query CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE");
+        if (u != 0)
+        {
+            log_error("CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES returned 0 but "
+                      "CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE returned a "
+                      "non-zero value\n");
+            return TEST_FAIL;
+        }
+
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_MAX_ON_DEVICE_QUEUES,
+                                sizeof(u), &u, NULL);
+        test_error(error, "Unable to query CL_DEVICE_MAX_ON_DEVICE_QUEUES");
+        if (u != 0)
+        {
+            log_error("CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES returned 0 but "
+                      "CL_DEVICE_MAX_ON_DEVICE_QUEUES returned a "
+                      "non-zero value\n");
+            return TEST_FAIL;
+        }
+
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_MAX_ON_DEVICE_EVENTS,
+                                sizeof(u), &u, NULL);
+        test_error(error, "Unable to query CL_DEVICE_MAX_ON_DEVICE_EVENTS");
+        if (u != 0)
+        {
+            log_error("CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES returned 0 but "
+                      "CL_DEVICE_MAX_ON_DEVICE_EVENTS returned a "
+                      "non-zero value\n");
+            return TEST_FAIL;
+        }
+
+        // clGetCommandQueueInfo, passing CL_QUEUE_SIZE or
+        // CL_QUEUE_DEVICE_DEFAULT
+        // Returns 0 or NULL if the device associated with command_queue does
+        // not support On-Device Queues.
+
+        error =
+            clGetCommandQueueInfo(queue, CL_QUEUE_SIZE, sizeof(u), &u, NULL);
+        // TODO: is this a valid query?  See:
+        // https://github.com/KhronosGroup/OpenCL-Docs/issues/402
+        // test_error(error, "Unable to query CL_QUEUE_SIZE");
```

Comment:
I think we probably want this to fail. I've commented on the spec issue.

Reply:
Sounds good.  Do you want to wait to merge these changes until the spec issue is resolved?

Reply:
Not necessarily but we should create a new issue if we're merging this PR ahead of resolving the spec issue so we don't lose track of this.